### PR TITLE
fix: Lotus Gateway: Add missing methods - master

### PR DESCRIPTION
**Pull request by ArseniiPetrovich**
Inspired by #10417 discussions
## Related Issues

#10402 

## Proposed Changes

Add 4 methods to Lotus Gateway:
- StateNetworkName
- MpoolGetNonce
- StateCall
- StateDecodeParams

## Additional Info

Required for Glif.io apps to work properly

## Checklist

Before you mark the PR ready for review, please make sure that:

- [x] Commits have a clear commit message.
- [x] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [ ] New features have usage guidelines and / or documentation updates in
  - [ ] [Lotus Documentation](https://lotus.filecoin.io)
  - [ ] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

**Review by arajasek — COMMENTED** [review]
LGTM, thanks for the PR!

File: api/proxy_gen.go [hunk 1]
```diff
@@ -45,6 +35,14 @@ import (
...
 	"github.com/filecoin-project/lotus/storage/sealer/sealtasks"
 	"github.com/filecoin-project/lotus/storage/sealer/storiface"
+	"github.com/google/uuid"
+	"github.com/ipfs/go-cid"
+	blocks "github.com/ipfs/go-libipfs/blocks"
+	"github.com/libp2p/go-libp2p/core/metrics"
+	"github.com/libp2p/go-libp2p/core/network"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/libp2p/go-libp2p/core/protocol"
+	"golang.org/x/xerrors"
```
Comment (magik6k) [inline api/proxy_gen.go:45]: Looks like you didn't run full `make gen`

Comment (ArseniiPetrovich) [reply api/proxy_gen.go]: @magik6k fixed ❤️

**Review by raulk — APPROVED** [review]
This LGTM, but @arajasek and @jennijuju should review.

**Comment by raulk** [comment]
Adding backport label since @ArseniiPetrovich wants this on Hyperspace too, I believe.

**Review by magik6k — APPROVED** [review]

**Pull request merged**